drivers/sensor/mchp_tach_xec/Kconfig

```diff
@@ -0,0 +1,34 @@
+# Microchip mec150x tachometer sensor configuration options
+
+# Copyright (c) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config TACH_XEC
+	bool "XEC Tachometer sensor"
+	depends on SOC_FAMILY_MEC
+	help
+	  Enable the Microchip XEC tachometer sensor.
+
+if TACH_XEC
+
+config TACH_XEC_0
+	bool "TACH 0"
+	 help
+	¦ Enable TACH instance 0.
+
+config TACH_XEC_1
+	bool "TACH 1"
+	help
+	¦ Enable TACH instance 1.
+
+config TACH_XEC_2
+	bool "TACH 2"
+	help
+	¦ Enable TACH instance 2.
+
+config TACH_XEC_3
+	bool "TACH 3"
+	help
+	¦ Enable TACH instance 3.
```

Comment:
Please remove these instance configs and use DT_INST macros in the driver instead.

Reply:
I don't understand this comment. I'm already using DT_INST in this new version please review again

Reply:
I think the idea is to avoid using Kconfig to enable each instances, but instead do `#ifdef DT_INST_MICROCHIP_XEC_TACH` instead of `#ifdef CONFIG_TACH_XEC_0` in the driver. So each `DEVICE_AND_API_INIT()` is wrapped with DT_INST instead of CONFIG.

Reply:
@galak I understand the trend to eliminate dts_fixup.h. But is there any document or discussion about eliminating Kconfig for turning on/off device instances ?  All the PRs I've seen are only replacing mixup macros for DT_INST_. Can you please review my PR?